src/main/java/net/oauth/jsontoken/JsonTokenParser.java

```diff
@@ -257,11 +253,11 @@ public boolean issuedAtIsValid(JsonToken jsonToken, Instant now) {
    * @throws IllegalStateException if tokenString is not a properly formatted JWT
    */
   private String[] splitTokenString(String tokenString) {
-    String[] pieces = tokenString.split(Pattern.quote(JsonTokenUtil.DELIMITER));
-    if (pieces.length != 3) {
+    List<String> pieces = Splitter.on(JsonTokenUtil.DELIMITER).limit(4).splitToList(tokenString);
+    if (pieces.size() != 3) {
       throw new IllegalStateException("Expected JWT to have 3 segments separated by '" + 
-          JsonTokenUtil.DELIMITER + "', but it has " + pieces.length + " segments");
+          JsonTokenUtil.DELIMITER + "', but it has " + pieces.size() + " segments");
```

Comment:
This error message could be inaccurate if you're limiting the splitter to 4.

Reply:
I changed it back to not having a limit, since knowing the exact length could be useful. My reasoning for having a limit of 4 was to stop parsing the token when it sees that it is already above the allowed limit. Would it make more sense to have a more general error message with the limit or have a more accurate message without the limit?

Reply:
Keeping the error message the same as before SGTM.